src/clang-cpp-frontend/clang_cpp_convert.cpp

```diff
@@ -666,10 +705,22 @@ bool clang_cpp_convertert::get_constructor_call(
   // Try to get the object that this constructor is constructing
   auto it = ASTContext->getParents(constructor_call).begin();
 
-  //exprt object;
   const clang::Decl *objectDecl = it->get<clang::Decl>();
+  if(!objectDecl)
+  {
+    address_of_exprt tmp_expr;
+    tmp_expr.type() = pointer_typet();
+    tmp_expr.type().subtype() = type;
+
+    exprt new_object("new_object");
+    new_object.set("#lvalue", true);
+    new_object.type() = type;
 
-  //call.arguments().push_back(object);
+    tmp_expr.operands().resize(0);
```

Comment:
What happens if we have a constructor with arguments? Will the resize(0) delete them?

Reply:
> Quick question: if you replace the new object with a malloc, do we still generate the valid_object call?

@mikhailramalho What do you mean by `generate the valid_object call`?

Reply:
No,  it will not delete them. `tmp_expr` refers to the `this` argument for constructor. Other arguments are added in the for loop followed by (see line 714 below).